Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/storage/retention/CheckInCleaner.kt

```diff
@@ -17,18 +16,12 @@ class CheckInCleaner @Inject constructor(
 
     suspend fun cleanUp() {
         Timber.d("Starting to clean up stale check-ins.")
-        val retentionThreshold = (timeStamper.nowUTC.seconds - RETENTION_SECONDS)
         val checkInsToDelete = checkInRepository.allCheckIns.first()
-            .filter {
-                it.checkInEnd.seconds < retentionThreshold
+            .filter { checkIn ->
+                isOutOfRetention(checkIn, timeStamper)
             }
```

Comment:
Before filtering please get the time now and pass the instance , with every checkin in a new instant is retrieved in `filter`

Reply:
good point. Fixed.